internal/state/match.go

```diff
@@ -18,16 +18,23 @@ type Match struct {
 	Atts      imap.FlagSet
 }
 
+type MatchMailbox struct {
```

Comment:
Could you make this private (`matchMailbox`) so as to not confuse with the other public "pseudo-mailbox" (`AppendOnlyMailbox`)?

Reply:
Done